main.tf

```
@@ -1,4 +1,14 @@
 resource "aws_iam_saml_provider" "this" {
-  name                   = var.saml_provider_name
-  saml_metadata_document = var.saml_provider_metadata
+  count = var.iam_identity_provider.saml != null ? 1 : 0
+
+  name                   = var.iam_identity_provider.saml.provider_name
+  saml_metadata_document = var.iam_identity_provider.saml.metadata_document
```

May as well send tags through also...

```suggestion
  saml_metadata_document = var.iam_identity_provider.saml.metadata_document
  tags                   = var.iam_identity_provider.saml.tags
```

---

added tags